tracer.go

```diff
@@ -283,6 +287,9 @@ func (t *Tracer) Extract(
 func (t *Tracer) Close() error {
 	t.reporter.Close()
 	t.sampler.Close()
+	if mgr, ok := t.baggageRestrictionManager.(io.Closer); ok {
```

Comment:
note that your remote manager is not an `io.Closer` : `func (m *RestrictionManager) Close() {` (no error returned)

Should add `var _ io.Closer = new(RemoteBaggageRestrictionManager)` in its test file.